pkgs/os-specific/linux/lenovo-legion/default.nix

```diff
@@ -0,0 +1,42 @@
+{ lib, fetchurl, stdenv, kernel, bash, pkgs }:
+
+stdenv.mkDerivation {
+  pname = "lenovo-legion";
+  inherit (pkgs.lenovo-legion) version src;
+
+  hardeningDisable = [ "pic" ];
+
+  preConfigure = ''
+    sed -i -e '/depmod/d' ./kernel_module/Makefile
+  '';
+
+  makeFlags = kernel.makeFlags ++ [
+    "SHELL=${bash}/bin/bash"
+    "KERNELVERSION=${kernel.modDirVersion}"
+    "KSRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+    "INSTALLDIR=${placeholder "out"}/lib/modules/${kernel.modDirVersion}"
+    "MODDESTDIR=${placeholder "out"}/${kernel.modDirVersion}/kernel/drivers/platform/x86"
+    "DKMSDIR=${placeholder "out"}/lib/modules/${kernel.modDirVersion}/misc"
+  ];
+
+  nativeBuildInputs = kernel.moduleBuildDependencies;
+
+  preBuild = "cd kernel_module";
```

Comment:
`sourceRoot` would be more idiomatic.

Reply:
Looks like we need to do something like this, not sure if it's better?
```nix 
  postUnpack = ''
    sourceRoot="$sourceRoot/kernel_module"
  '';
  ```

Reply:
I would next build with -K/--keep-failed and inspect the files to see what's happening.